examples/alerts/alerts.md

```diff
@@ -0,0 +1,143 @@
+    dashboard: RULE_DASHBOARD
+```
+
+And some alerts running in thanos rule:
+
+```
+- alert: ThanosRuleGrpcErrorRate
+  expr: rate(grpc_server_handled_total{grpc_code=~"Unknown|ResourceExhausted|Internal|Unavailable",app="thanos-rule"}[5m]) > 0
+  for: 5m
+  labels:
+    team: TEAM
+  annotations:
+    summary: Thanos Rule is returning Internal/Unavailable errors
+    impact: Recording Rules are not working
+    action: Check {{ $labels.kubernetes_pod_name }} pod logs in {{ $labels.kubernetes_namespace}} namespace
+    dashboard: RULE_DASHBOARD
+```
+
+## Store Gateway
+
+```
+- alert: ThanosStoreGrpcErrorRate
+  expr: rate(grpc_server_handled_total{grpc_code=~"Unknown|ResourceExhausted|Internal|Unavailable",app="thanos-store"}[5m]) > 0
+  for: 5m
+  labels:
+    team: TEAM
+  annotations:
+    summary: Thanos Store is returning Internal/Unavailable errors 
+    impact: Long Term Storage Prometheus queries are failing
+    action: Check {{ $labels.kubernetes_pod_name }} pod logs in {{ $labels.kubernetes_namespace}} namespace
+    dashboard: GATEWAY_URL
+```
+
+## Sidecar
+
+```
+- alert: ThanosSidecarPrometheusDown
+  expr: thanos_sidecar_prometheus_up{name="prometheus"} == 0
+  for: 5m
+  labels:
+    team: TEAM
+  annotations:
+    summary: Thanos Sidecar cannot connect to Prometheus
+    impact: Prometheus configuration is not being refreshed
+    action: Check {{ $labels.kubernetes_pod_name }} pod logs in {{ $labels.kubernetes_namespace}} namespace
+    dashboard: SIDECAR_URL
+- alert: ThanosSidecarBucketOperationsFailed
+  expr: rate(thanos_objstore_bucket_operation_failures_total{name="prometheus"}[5m]) > 0
+  for: 5m
+  labels:
+    team: TEAM
+  annotations:
+    summary: Thanos Sidecar bucket operations are failing
+    impact: We will lose metrics data if not fixed in 24h
+    action: Check {{ $labels.kubernetes_pod_name }} pod logs in {{ $labels.kubernetes_namespace}} namespace
+    dashboard: SIDECAR_URL
+- alert: ThanosSidecarGrpcErrorRate
+  expr: rate(grpc_server_handled_total{grpc_code=~"Unknown|ResourceExhausted|Internal|Unavailable",name="prometheus"}[5m]) > 0
+  for: 5m
+  labels:
+    team: TEAM
+  annotations:
+    summary: Thanos Sidecar is returning Internal/Unavailable errors 
+    impact: Prometheus queries are failing
+    action: Check {{ $labels.kubernetes_pod_name }} pod logs in {{ $labels.kubernetes_namespace}} namespace
+    dashboard: SIDECAR_URL
+```
+
+## Query
+
+```
+- alert: ThanosQueryGrpcErrorRate
+  expr: rate(grpc_server_handled_total{grpc_code=~"Unknown|ResourceExhausted|Internal|Unavailable",name="prometheus"}[5m]) > 0
+  for: 5m
+  labels:
+    team: TEAM
+  annotations:
+    summary: Thanos Query is returning Internal/Unavailable errors 
+    impact: Grafana is not showing metrics
+    action: Check {{ $labels.kubernetes_pod_name }} pod logs in {{ $labels.kubernetes_namespace}} namespace
+    dashboard: QUERY_URL
```

Comment:
what about query not having access to static stores? or query seeing unhealthy stores.

Reply:
I am not sure which metrics to use for that, but I typically like to alert when things really break, so I think GRPC error rate is a good proxy for that. And from there you can debug via dashboard.